File: 2019/04/python_day04/day04_test.py
#!/usr/bin/env python
from unittest import TestCase, main
from day04 import is_password, is_password2, solve, solve2


class Day04TestCase(TestCase):
    def test_is_password(self):
        self.assertEqual(is_password(111111), True)
        self.assertEqual(is_password(223450), False)
        self.assertEqual(is_password(123789), False)
        self.assertEqual(is_password(123444), True)

    def test_is_password2(self):
        self.assertEqual(is_password2(111111), False)
        self.assertEqual(is_password2(223450), False)
        self.assertEqual(is_password2(123789), False)
        self.assertEqual(is_password2(112233), True)
        self.assertEqual(is_password2(123444), False)
        self.assertEqual(is_password2(111122), True)
        self.assertEqual(is_password2(111123), False)

    def test_solve(self):
        want = 771
        got = solve(200000, 300000)
        self.assertEqual(want, got)

    def test_solve2(self):
        want = 546
        got = solve2(200000, 300000)
        self.assertEqual(want, got)


if __name__ == "__main__":
    main()

File: 2019/04/go_day04/day04_test.go
// Package main provides ...
package main

import (
	"strconv"
	"testing"
)

func TestSolve(t *testing.T) {
	t.Run("It works", func(t *testing.T) {
		got := Solve(200000, 300000)
		want := 771
		if got != want {
			t.Errorf("got %v want %v", got, want)
		}
	})
}
func TestSolve2(t *testing.T) {
	t.Run("It works", func(t *testing.T) {
		got := Solve2(200000, 300000)
		want := 546
		if got != want {
			t.Errorf("got %v want %v", got, want)
		}
	})
}

func TestIsPassword(t *testing.T) {
	tests := []struct {
		input  int
		output bool
	}{
		{111111, true},
		{223450, false},
		{123789, false},
		{123444, true},
	}

	for _, test := range tests {
		name := strconv.Itoa(test.input)
		t.Run(name, func(t *testing.T) {
			want := test.output
			got := IsPassword(test.input)
			if got != want {
				t.Errorf("got %v want %v", got, want)
			}
		})
	}
}

func TestIsPassword2(t *testing.T) {
	tests := []struct {
		input  int
		output bool
	}{
		{111111, false},
		{223450, false},
		{123789, false},
		{123444, false},
		{112233, true},
		{123444, false},
		{111122, true},
		{111123, false},
	}

	for _, test := range tests {
		name := strconv.Itoa(test.input)
		t.Run(name, func(t *testing.T) {
			want := test.output
			got := IsPassword2(test.input)
			if got != want {
				t.Errorf("got %v want %v", got, want)
			}
		})
	}
}
